spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/MLPredictorConfig.java

```diff
@@ -0,0 +1,175 @@
+  @Description("A comma-separated sequence of fields that needs to be excluded from being used " +
+    "in classification/prediction.")
+  protected String featureFieldsToExclude;
+
+  @Description("The field on which to set the prediction. It will be of type double.")
+  protected String predictionField;
+
+  public MLPredictorConfig() {
+  }
+
+  public MLPredictorConfig(String fileSetName, String path, @Nullable String featureFieldsToInclude,
+                         @Nullable String featureFieldsToExclude, String predictionField) {
+    this.fileSetName = fileSetName;
+    this.path = path;
+    this.featureFieldsToInclude = featureFieldsToInclude;
+    this.featureFieldsToExclude = featureFieldsToExclude;
+    this.predictionField = predictionField;
+  }
+
+  /**
+   * Validate the config parameters for the spark sink class.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training.
+   * @param featuresToExclude features to be excluded when training.
+   * @param predictionField   field containing the label values.
+   */
+  public static void validateConfigParameters(Schema inputSchema, @Nullable String featuresToInclude,
+                                              @Nullable String featuresToExclude, String predictionField,
+                                              @Nullable String cardinalityMapping) {
+    if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
+      throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
+                                           "Please specify fields for one.");
+    }
+
+    Schema.Field prediction = inputSchema.getField(predictionField);
+    if (prediction != null) {
+      throw new IllegalArgumentException(String.format("Prediction field %s should not be present in the input schema.",
+                                                       predictionField));
+    }
+
+    getCategoricalFeatureInfo(inputSchema, featuresToInclude, featuresToExclude, predictionField, cardinalityMapping);
+  }
+
+  /**
+   * Get the feature list of the features that have to be used for training/prediction depending on the
+   * featuresToInclude or featuresToInclude list.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param predictionField   field containing the label values.
+   * @return feature list to be used for training/prediction.
+   */
+  static Map<String, Integer> getFeatureList(Schema inputSchema, @Nullable String featuresToInclude,
+                                             @Nullable String featuresToExclude, String predictionField) {
+    if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
```

Comment:
you can still call that utils class from here instead of copying all the code. Otherwise there is a lot of code duplication between this and the trainer config.